rocksdb: 6.29.3 -> 7.3.1

Mo0nbase opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Mo0nbase committed:
rocksdb: 6.29.3 -> 7.3.1

viraptor commented:
Result of `nixpkgs-review pr 179399` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>matrix-conduit</li>
    <li>sortmerna</li>
  </ul>
</details>
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>fulcrum</li>
    <li>haskellPackages.corenlp-parser</li>
    <li>haskellPackages.rocksdb-haskell</li>
    <li>haskellPackages.rocksdb-haskell-jprupp</li>
    <li>haskellPackages.rocksdb-query</li>
    <li>nimPackages.rocksdb</li>
    <li>nimPackages.spry</li>
    <li>rocksdb</li>
    <li>rocksdb_lite</li>
  </ul>
</details>

```
[23/633] Building CXX object CMakeFiles/rocksdb-shared.dir/cache/fast_lru_cache.cc.o
FAILED: CMakeFiles/rocksdb-shared.dir/cache/fast_lru_cache.cc.o
/nix/store/r6hlkiia72wxql9q6pcb61rhjz7hrkwr-clang-wrapper-11.1.0/bin/clang++ -DBZIP2 -DHAVE_FULLFSYNC -DLZ4 -DOS_MACOSX -DROCKSDB_LIB_IO_POSIX -DROCKSDB_NO_DYNAMIC_EXTENSION -DROCKSDB_PLATFORM_POSIX -DSNAPPY -DZLIB -DZSTD -Drocksdb_shared_EXPORTS -I/tmp/nix-build-rocksdb-7.3.1.drv-0/source -I/tmp/nix-build-rocksdb-7.3.1.drv-0/source/include -isystem /nix/store/x5jq9b59k12s25jsl986j17s99klrpqc-snappy-1.1.9-dev -W -Wextra -Wall -pthread -Wsign-compare -Wshadow -Wno-unused-parameter -Wno-unused-variable -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-invalid-offsetof -fno-omit-frame-pointer -momit-leaf-frame-pointer -Werror -O3 -DNDEBUG -DROCKSDB_USE_RTTI -fPIC -std=gnu++17 -MD -MT CMakeFiles/rocksdb-shared.dir/cache/fast_lru_cache.cc.o -MF CMakeFiles/rocksdb-shared.dir/cache/fast_lru_cache.cc.o.d -o CMakeFiles/rocksdb-shared.dir/cache/fast_lru_cache.cc.o -c /tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/fast_lru_cache.cc
In file included from /tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/fast_lru_cache.cc:10:
/tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/fast_lru_cache.h:158:3: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
  ~LRUCacheShard() override = default;
  ^
/tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/fast_lru_cache.cc:457:18: note: in defaulted destructor for 'rocksdb::fast_lru_cache::LRUCacheShard' first required here
      shards_[i].~LRUCacheShard();
                 ^
/tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/fast_lru_cache.h:158:3: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
  ~LRUCacheShard() override = default;
  ^
1 error generated.
[24/633] Building CXX object CMakeFiles/rocksdb-shared.dir/cache/compressed_secondary_cache.cc.o
[25/633] Building CXX object CMakeFiles/rocksdb-shared.dir/cache/lru_cache.cc.o
FAILED: CMakeFiles/rocksdb-shared.dir/cache/lru_cache.cc.o
/nix/store/r6hlkiia72wxql9q6pcb61rhjz7hrkwr-clang-wrapper-11.1.0/bin/clang++ -DBZIP2 -DHAVE_FULLFSYNC -DLZ4 -DOS_MACOSX -DROCKSDB_LIB_IO_POSIX -DROCKSDB_NO_DYNAMIC_EXTENSION -DROCKSDB_PLATFORM_POSIX -DSNAPPY -DZLIB -DZSTD -Drocksdb_shared_EXPORTS -I/tmp/nix-build-rocksdb-7.3.1.drv-0/source -I/tmp/nix-build-rocksdb-7.3.1.drv-0/source/include -isystem /nix/store/x5jq9b59k12s25jsl986j17s99klrpqc-snappy-1.1.9-dev -W -Wextra -Wall -pthread -Wsign-compare -Wshadow -Wno-unused-parameter -Wno-unused-variable -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-invalid-offsetof -fno-omit-frame-pointer -momit-leaf-frame-pointer -Werror -O3 -DNDEBUG -DROCKSDB_USE_RTTI -fPIC -std=gnu++17 -MD -MT CMakeFiles/rocksdb-shared.dir/cache/lru_cache.cc.o -MF CMakeFiles/rocksdb-shared.dir/cache/lru_cache.cc.o.d -o CMakeFiles/rocksdb-shared.dir/cache/lru_cache.cc.o -c /tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/lru_cache.cc
In file included from /tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/lru_cache.cc:10:
/tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/lru_cache.h:291:11: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
  virtual ~LRUCacheShard() override = default;
          ^
/tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/lru_cache.cc:674:18: note: in defaulted destructor for 'rocksdb::lru_cache::LRUCacheShard' first required here
      shards_[i].~LRUCacheShard();
                 ^
/tmp/nix-build-rocksdb-7.3.1.drv-0/source/cache/lru_cache.h:291:11: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
  virtual ~LRUCacheShard() override = default;
          ^
1 error generated.
```

veprbl committed:
rocksdb: fix on darwin

error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer

viraptor reviewed (approved):
Result of `nixpkgs-review pr 179399` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>matrix-conduit</li>
    <li>sortmerna</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>blockbook</li>
    <li>electrs</li>
    <li>fulcrum</li>
    <li>haskellPackages.corenlp-parser</li>
    <li>haskellPackages.rocksdb-haskell</li>
    <li>haskellPackages.rocksdb-haskell-jprupp</li>
    <li>haskellPackages.rocksdb-query</li>
    <li>nimPackages.rocksdb</li>
    <li>nimPackages.spry</li>
    <li>rocksdb</li>
    <li>rocksdb_6_23</li>
    <li>rocksdb_lite</li>
  </ul>
</details>

veprbl commented:
Result of `nixpkgs-review pr 179399` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>matrix-conduit</li>
    <li>sortmerna</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>electrs</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>blockbook</li>
    <li>fulcrum</li>
    <li>nimPackages.rocksdb</li>
    <li>nimPackages.spry</li>
    <li>rocksdb</li>
    <li>rocksdb_6_23</li>
    <li>rocksdb_lite</li>
  </ul>
</details>
